nemo/collections/nlp/models/token_classification/punctuation_capitalization_lexical_audio_model.py

```diff
@@ -0,0 +1,260 @@
+
+        self.audio_encoder = nemo_asr.models.ASRModel.from_pretrained(
+            cfg.audio_encoder.pretrained_model, override_config_path=audio_cfg
+        )
+        if cfg.audio_encoder.adapter.get('enable', False):
+            with open_dict(cfg):
+                cfg.audio_encoder.adapter.config.in_features = self.audio_encoder.cfg.decoder.feat_in
+            self.audio_encoder.add_adapter(name='audio_adapter', cfg=cfg.audio_encoder.adapter.config)
+            self.audio_encoder.set_enabled_adapters(enabled=True)
+            self.audio_encoder.freeze()
+            self.audio_encoder.unfreeze_enabled_adapters()
+
+        self.fusion = TransformerDecoder(
+            num_layers=cfg.audio_encoder.fusion.num_layers,
+            hidden_size=self.bert_model(**self.bert_model.input_example()[0]).size()[-1],
+            inner_size=cfg.audio_encoder.fusion.inner_size,
+            num_attention_heads=cfg.audio_encoder.fusion.num_attention_heads,
+        )
+        self.audio_proj = Linear(
+            self.audio_encoder.cfg.decoder.feat_in, self.bert_model(**self.bert_model.input_example()[0]).size()[-1]
+        )
+
+        if cfg.audio_encoder.freeze.get('is_enabled', False):
+            for param in self.audio_encoder.parameters():
+                param.requires_grad = False
+            for i in range(cfg.audio_encoder.fusion.get('num_layers')):
+                self.audio_encoder.add_module(
+                    f'conf_encoder_{i}',
+                    ConformerLayer(
+                        d_model=cfg.audio_encoder.freeze.get('d_model'), d_ff=cfg.audio_encoder.freeze.get('d_ff')
+                    ),
+                )
+
+        if cfg.get('restore_lexical_encoder_from', None) and not self._is_model_being_restored():
+            self.bert_model = (
+                PunctuationCapitalizationModel.restore_from(cfg.restore_lexical_encoder_from)
+                .to(self.device)
+                .bert_model
+            )
+
+        del self.audio_encoder.decoder
+        del self.audio_encoder._wer
+        del self.audio_encoder.loss
+
+        if cfg.get('use_weighted_loss', False):
+            punct_freq = torch.tensor(
+                list(self.train_dataloader().dataset.punct_label_frequencies.values()), dtype=torch.float
+            )
+            punct_weight = 1 - (punct_freq - punct_freq.min()) / punct_freq.max()
+
+            capit_freq = torch.tensor(
+                list(self.train_dataloader().dataset.capit_label_frequencies.values()), dtype=torch.float
+            )
+            capit_weight = 1 - (capit_freq - capit_freq.min()) / capit_freq.max()
+
+            self.loss_punct = CrossEntropyLoss(logits_ndim=3, weight=punct_weight)
+            self.loss_capit = CrossEntropyLoss(logits_ndim=3, weight=capit_weight)
+        else:
+            self.loss_punct = self.loss
+            self.loss_capit = self.loss
+
+        self.set_max_audio_length(1024)
```

Comment:
Because it does not matter in this case. It's going to be updated at first call of `forward` and not used before.
It makes sense in ASR config but not here as it's only purpose is keep track of max length for mask creation